### PR TITLE
chore: minor tuning of GHA build

### PR DESCRIPTION
**Pull request by ssbarnea**
## Description

This should address error seen on https://github.com/streetsidesoftware/cspell-dicts/actions/runs/11125090808/job/30912037317?pr=3446#step:3:67 which should have not happen, especially as the package.json and its lock were not touched by that change.

## References

- _Any source references._

## Checklist

- [x] By submitting this pull-request, you agree to follow our [Code of Conduct](https://github.com/streetsidesoftware/cspell-dicts/blob/main/CODE_OF_CONDUCT.md)
- [x] Verify that the title starts with the correct prefix:
  - `fix:` - for minor changes like adding words or fixing spelling issues.
  - `feat:` - for a significant change like adding a whole new set of words to a dictionary.
  - `feat!:` - for breaking changes, like file format or licensing changes.
  - `chore:` - for changes that do not impact the content of dictionaries.

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1 @@
+nodejs 22.9.0
```
Comment (Jason3S) [inline .tool-versions:1]: Do you know which tools us this file?

File: .github/actions/setup/action.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ inputs:
   node-version:
     required: false
     description: The version of Node to use.
-    default: '18.x'
+    default: '22.x'
```
Comment (Jason3S) [inline .github/actions/setup/action.yaml:7]: It should not make much difference, but I had been keeping this at the same level as the CSpell support level, which is 18+.

File: dictionaries/docker/source/Dockerfile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # syntax=docker/dockerfile:1
-FROM node:14.21.3-alpine
+FROM node:22.9.0-alpine
```
Comment (Jason3S) [inline dictionaries/docker/source/Dockerfile:2]: I didn't even realize this file was there. I think it was added as a `sample` but incorrectly put in `source`. It isn't used.

**Review by Jason3S — COMMENTED** [review]
Thank you for the heads up on the broken build.

File: .tool-versions (hunk 1 above)
Comment (ssbarnea) [reply .tool-versions:1]: Originally it was comes from `asdf` but the great thing is that the official https://github.com/actions/setup-node does support it too, making so easy to control the version of node being used locally and on CI too. Noted It does not override a manually configured version in the action.

Personally, I would configure cspell-dicts repo in particular to use a single version of node. That repo does not really need testing with multiple versions as it builds only dictionaries. For the main cspell repo, that is another story.

Controlling the version should lower the chance of getting surprises, aka more stable CI/CD pipelines.

File: .github/actions/setup/action.yaml (hunk 2 above)
Comment (ssbarnea) [reply .github/actions/setup/action.yaml:7]: I should put 18.x in the .tool-versions file in that case? I can even configure the action to take it from there.

Comment (Jason3S) [reply .github/actions/setup/action.yaml:7]: Yes, that would be great.

**Comment by ssbarnea** [comment]
I made it draft as I will use to modify the actions to make use of the file.

One thing that I observed is that pnpm has the bad habit of updating the lockfile by default on install, that being different than npm and yarn, which need an extra argument to refresh the lock. pnpm goes the other way around and requires the switch to make it use the file.

Based on version of node being used (18 or 20), it seems that it updates two types files, very unlikely to really break the code but enough to cause churn.

I guess that we need to be sure that the lock file is produced with minimal supported version of node.

File: .tool-versions (hunk 1 above)
Comment (Jason3S) [reply .tool-versions]: @ssbarnea,

Cool! I didn't know that.

**Comment by Jason3S** [comment]
> I made it draft as I will use to modify the actions to make use of the file.
> 
> One thing that I observed is that pnpm has the bad habit of updating the lockfile by default on install, that being different than npm and yarn, which need an extra argument to refresh the lock. pnpm goes the other way around and requires the switch to make it use the file.
> 
> Based on version of node being used (18 or 20), it seems that it updates two types files, very unlikely to really break the code but enough to cause churn.
> 
> I guess that we need to be sure that the lock file is produced with minimal supported version of node.

There is a workflow (update-dependencies) that will fix the `pnpm-lock.yaml` file. There is a bit of churn, but it settles on the workflow version.

**Pull request merged**